server/redis.repository.js

```diff
@@ -0,0 +1,79 @@
+const redis = require("redis");
+const { promisify } = require("util");
+const client = redis.createClient();
+const hgetAsync = promisify(client.hget).bind(client);
+const hincrAsync = promisify(client.hincrby).bind(client);
+const hsetAsync = promisify(client.hset).bind(client);
+
+/**
+ * Creates a room key in Redis and sets:
+ *      1. the number of movies
+ *      2. the number of guests
+ *      3. the number of total swipes to 0
+ * 
+ * @param {String} roomId - The room ID.
+ * @param {Integer} numGuests - The number of guests in the room, including the host
+ * @param {Object[]} movies - The list of movies with the respective movie Ids
+ * @returns {Integer} x - Number of fields sets (No errors if greater than 0)
+ */
+const initializeRoom = async (roomId, numGuests, movies) => {
+    let roomArgs = ["numUsers", numGuests, "total_swipes", 0, "numMovies", movies.length];
+    movies.forEach(element => {
+        roomArgs.push(element.id);
+        roomArgs.push(0);
+    });
+
+    return await hsetAsync(roomId, roomArgs);
+}
+
+/**
+ * Increments total_swipes and number of likes for movie. Also checks if match or if swiping completed. 
+ * 
+ * @param {String} roomId - The room ID.
+ * @param {Integer} movie - Movie ID
+ * @returns {Integer} 0 - If match found for movie ID passed in.
+ * @returns {Integer} 1 - If no match found but swiping completed. 
+ * @returns {Integer} -1 - If no match found and swiping not completed. 
+ */
+const likeEvent = async (roomId, movie) => {
+    const total_swipes = await hincrAsync(roomId, "total_swipes", 1);
+    const movie_likes = await hincrAsync(roomId, movie, 1);
+
```

Comment:
I believe that the race condition where 2 or more movies that match at the same time (i.e. reach movie_likes = num_guests) will still exist since the num_guests call for different rooms could potentially return in a non-ordered sequence (movie 1 matches first, but movie 2 call returns first). 

However, this is unlikely to occur, and we haven't discussed if returning the first matched movie (accurate to within milliseconds) is a sprint or project priority, so for the purposes of this sprint I think that this implementation is sufficient. 

We should discuss in future meetings if this degree of accuracy is sufficient for the purposes of WeWatch, as I think avoiding race conditions like these may not be necessary for the purposes of our project, and we can forego a complicated transaction approach.

Reply:
+1 to this! thanks for bringing up the race condition Derek.

Reply:
Great find! We can discuss this during standup tomorrow and I will update this thread.

Reply:
Discussed during standup revealed that this race condition is okay based on the requirements/scope of this project.